src/sage/matrix/seymour_decomposition.pyx

```diff
@@ -902,6 +1016,44 @@ cdef class SubmatrixNode(DecompositionNode):
     pass
 
 
+cdef class DecompositionNodeIndexKey:
```

Comment:
perhaps make it `ElementKey` to connect to the CMR terminology?

Reply:
Sounds good!